File: src/main/java/org/goafabric/dbagentnew/persistence/PersonRepository.java
package org.goafabric.dbagentnew.persistence;

import org.goafabric.dbagentnew.persistence.entity.PersonEo;
import org.springframework.data.repository.CrudRepository;

import java.util.List;

public interface PersonRepository extends CrudRepository<PersonEo, String> {

    List<PersonEo> findByFirstNameIgnoreCase(String firstName);

    List<PersonEo> findByLastNameIgnoreCase(String lastName);

    List<PersonEo> findByAddressStreetContainsIgnoreCase(String street);

    List<PersonEo> findByAddressCityContainsIgnoreCase(String city);

}

File: src/main/kotlin/org/goafabric/llm/tool/persistence/PersonRepository.kt
package org.goafabric.llm.tool.persistence

import org.goafabric.llm.tool.persistence.entity.Person
import org.springframework.data.repository.CrudRepository

interface PersonRepository : CrudRepository<Person, String> {
    fun findByFirstNameIgnoreCase(firstName: String): MutableList<Person>

    fun findByLastNameIgnoreCase(lastName: String): MutableList<Person>

    fun findByAddressStreetContainsIgnoreCase(street: String): MutableList<Person>

    fun findByAddressCityContainsIgnoreCase(city: String): MutableList<Person>

    fun findByAllergyAllergyContainsIgnoreCase(allergy: String): MutableList<Person>
}

